### PR TITLE
keep track of the samples used while peak detection

### PR DESCRIPTION
**Pull request by rish9511**
For each group that gets formed during the peak detection, keep track of the samples that were used for it

When exporting the groups to csv, intensities of the samples should be displayed only if they were used while peak detection

Issue: #261

### CODE REVIEW

**Comment by chubukov** [comment]
@rish9511 Great idea. Have you considered the potential pitfalls if samples get deleted between peak detection and recall?

File: libmaven/PeakGroup.cpp [hunk 1]
```diff
@@ -237,7 +237,13 @@ vector<float> PeakGroup::getOrderedIntensityVector(vector<mzSample*>& samples, Q
 
     for( unsigned int j=0; j < samples.size(); j++) {
         sampleOrder[samples[j]]=j;
-        maxIntensity[j]=0;
+        // check if the the sample was used while peak detection or not
+        // samplesUsed is populate during peakDetection
+        auto it = std::find(std::begin(samplesUsed), std::end(samplesUsed), samples[j]);
+        if(it != std::end(samplesUsed))
```
Comment (shubhra-agrawal) [inline libmaven/PeakGroup.cpp]: We're setting maxIntensity[j] = -1 here when line 270 is doing the same thing later after normalization. Any particular reason for it?

Comment (rish9511) [inline libmaven/PeakGroup.cpp]: It's just a sanity check but not sure whether it's needed

**Comment by rish9511** [comment]
@chubukov 
Thanks for pointing out. We have taken care of it now. The vector that stores the used samples is made to remove the sample being deleted

Another thing I wanted to ask was, should the sample be exported to csv if it's deleted ? It's not exported as of now

**Comment by chubukov** [comment]
@rish9511 I would say no. If it's deleted, it should be safe to remove all associations with it from all the peakgroups.

**Pull request closed without merging**